tools/build_cdn.js

```diff
@@ -1,37 +1,93 @@
 const fs = require("fs").promises;
 const fss = require("fs");
 const glob = require("glob");
+const Terser = require("terser");
 const zlib = require('zlib');
 const { getLanguages } = require("./lib/language");
 const { filter } = require("./lib/dependencies");
 const config = require("./build_config");
 const { install, installCleanCSS, mkdir } = require("./lib/makestuff");
 const log = (...args) => console.log(...args);
 const { buildBrowserHighlightJS } = require("./build_browser");
-const { buildPackageJSON } = require("./build_node");
+const { buildPackageJSON, writePackageJSON } = require("./build_node");
+const { rollupCode } = require("./lib/bundling.js");
 const path = require("path");
 const bundling = require('./lib/bundling.js');
 
 async function installPackageJSON(options) {
-  await buildPackageJSON(options);
-  const json = require(`${process.env.BUILD_DIR}/package`);
+  const json = buildPackageJSON(options, {
+    ".": {
```

Comment:
Why would this matter for web usage?

Reply:
I see now that the npm `exports` has been drug along into the v11 CDN package unintentionally, but I'm not sure this is necessary at all.  There are two common uses cases for CDN:

- Purely a delivery system to get our files onto CDNs, where they are directly loaded/imported (most common)
- Via `npm` when someone really wants JUST the build assets to directly copy into their project

I'm not sure `exports` is necessary for either of these cases.

Thoughts?

Reply:
I don't think it's necessary indeed, but I thought it wouldn't hurt to keep it I think. It could be helpful for an hypothetical future tool that can generate an import map from a `package.json` `"exports"`, and humans can also read it to understand the structure of the files to understand which url they should import.